src/codegen/generators/expression_generator.rs

```diff
@@ -720,6 +726,20 @@ impl<'a, 'b> ExpressionCodeGenerator<'a, 'b> {
             let variable_index_entry = self
                 .index
                 .find_variable(Some(type_name), &[name.to_string()])
+                .or_else(|| {
+                    let annotation = self.annotations.get(context)?;
+                    match annotation {
+                        StatementAnnotation::Variable {
+                            resulting_type: _,
+                            qualified_name,
+                        } => {
+                            let qualifier = &qualified_name[..qualified_name.rfind('.')?];
+                            self.index
+                                .find_variable(Some(qualifier), &[name.to_string()])
+                        }
```

Comment:
That's where I had to introduce a little hack. I'm trimming off the variable name from the full qualified reference string here to be able to use it in `find_variable()`. This should be fixed in a future PR where full qualified references will be replaced by some sort of struct.